spark/src/main/kotlin/com/adevinta/spark/components/divider/Divider.kt

```diff
@@ -21,92 +21,215 @@
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
 import androidx.compose.material3.DividerDefaults
 import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.style.TextAlign
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.tooling.preview.PreviewParameter
-import androidx.compose.ui.unit.Dp
-import com.adevinta.spark.ExperimentalSparkApi
+import androidx.compose.ui.unit.dp
 import com.adevinta.spark.PreviewTheme
 import com.adevinta.spark.SparkTheme
+import com.adevinta.spark.components.spacer.HorizontalSpacer
+import com.adevinta.spark.components.text.Text
 import com.adevinta.spark.tokens.dim3
 import com.adevinta.spark.tools.preview.ThemeProvider
 import com.adevinta.spark.tools.preview.ThemeVariant
-import androidx.compose.material3.HorizontalDivider as MaterialDivider
+import androidx.compose.material3.HorizontalDivider as MaterialHorizontalDivider
 import androidx.compose.material3.VerticalDivider as MaterialVerticalDivider
 
-@ExperimentalSparkApi
-@Composable
-internal fun SparkDivider(
-    modifier: Modifier = Modifier,
-    color: Color = SparkTheme.colors.outline,
-    thickness: Dp = DividerDefaults.Thickness,
-) {
-    MaterialDivider(
-        modifier,
-        thickness,
-        color,
-    )
+public enum class LabelHorizontalAlignment {
+    Start,
+    Center,
+    End,
 }
 
-/**
- * Spark divider.
- *
- * A divider is a thin line that groups content in lists and layouts.
- *
- * ![Divider image](https://developer.android.com/images/reference/androidx/compose/material3/divider.png)
- *
- * @param modifier the [Modifier] to be applied to this divider line.
- * @param color the color of the divider, we recommend to use either outline or onSurface dim 3.
- */
-@ExperimentalSparkApi
+public enum class LabelVerticalAlignment {
+    Top,
+    Center,
+    Bottom,
+}
+
+@Deprecated(
+    "Divider is deprecated",
+    ReplaceWith("HorizontalDivider()"),
+)
 @Composable
 public fun Divider(
     modifier: Modifier = Modifier,
     color: Color = SparkTheme.colors.onSurface.dim3,
 ) {
-    SparkDivider(
+    MaterialHorizontalDivider(
         modifier = modifier,
         thickness = DividerDefaults.Thickness,
         color = color,
     )
 }
 
-/**
- * Spark divider.
- *
- * A divider is a thin line that groups content in lists and layouts.
- *
- * ![Divider image](https://developer.android.com/images/reference/androidx/compose/material3/divider.png)
- *
- * @param modifier the [Modifier] to be applied to this divider line.
- * @param color the color of the divider, we recommend to use either outline or onSurface dim 3.
- */
-@ExperimentalSparkApi
+@Composable
+public fun HorizontalDivider(
```

Comment:
Let's use `Divider` as the horizontal divider otherwise it'll require our consumers to rename it everywhere for no real benefits since the signature is different from the previous one already…

Reply:
I don't prefer this approach,
Even tho, clients would do small changes,

The end goal we will have a unified approach like material design,
also we will follow the same pattern as Spacer.

Let's keep it like this.